src/pages-and-resources/PagesAndResources.jsx

```diff
@@ -46,6 +42,26 @@ function PagesAndResources({ courseId, intl }) {
           >
             <DiscussionsSettings courseId={courseId} />
           </PageRoute>
+          <PageRoute path={`${path}/:appId/settings`}>
+            {
+              ({ match, history }) => {
+                const SettingsComponent = React.lazy(async () => {
+                  try {
+                    // There seems to be an issue with eslint when using template strings here.
+                    return await import('./' + match.params.appId + '/Settings.jsx'); // eslint-disable-line
```

Comment:
this string concatenation can be fixed by changing the syntax to

  return await import(`./${  match.params.appId  }/Settings.jsx`);

Reply:
I am aware of that syntax, which is why I've left a comment to explain  that that doesn't work.

Reply:
What is the issue, can you explain a little more, also we have tried this syntax and ran the application, it seems to work for us.

Reply:
App setting files doesn't exist for all apps except `calculator` hence clicking settings icon is giving errors.
also
string concatenation is working in lazy loading import for components.

Reply:
@mehaknasir 
The issue is that eslint will essentially crash if you use a template string here. You can see more details on this bug here: https://github.com/babel/babel-eslint/issues/530

@awaisdar001 As mentioned in out previous discussion, I've removed all the components, and will put them in a separate PR to simplify review. This PR includes the base code and a single settings component to show it in use.